consensus/core/src/core.rs

```diff
@@ -519,6 +544,11 @@ impl Core {
             .collect()
     }
 
+    /// Returns the 1st leader of the round.
+    fn first_round_leader(&self, round: Round) -> AuthorityIndex {
+        self.leaders(round).first().unwrap().authority
```

Comment:
Minor for now, but this is guaranteed to work for the pipelined version only. Maybe use an "expect" instead with a message like "no leader exists, probably using in non-pipelined mode"

Reply:
I think we should remove the non-pipelined option. The `leaders_exist()` function will not work correctly for non-pipelined version either. And I suspect there will be a lot more issues.